crates/compression/src/context/compress.rs

```diff
@@ -0,0 +1,117 @@
+use std::collections::HashMap;
+
+use fuel_core_types::{
+    fuel_compression::{
+        CompressibleBy,
+        ContextError,
+        RegistryKey,
+    },
+    fuel_tx::{
+        input::PredicateCode,
+        Address,
+        AssetId,
+        CompressedUtxoId,
+        ContractId,
+        ScriptCode,
+        UtxoId,
+    },
+};
+
+use crate::{
+    compress::CompressDb,
+    eviction_policy::CacheEvictor,
+    ports::EvictorDb,
+    tables::{
+        PerRegistryKeyspace,
+        PostcardSerialized,
+        RegistryKeyspace,
+    },
+};
+
+pub struct CompressCtx<D> {
+    pub db: D,
+    pub cache_evictor: CacheEvictor,
+    /// Changes to the temporary registry, to be included in the compressed block header
+    pub changes: PerRegistryKeyspace<HashMap<RegistryKey, PostcardSerialized>>,
```

Comment:
Everywhere in the codebase, we heavily rely on the `PostcardSerialized` type and on the fact that we use poscard. Even in the database where it is up to the database to decide how to serialize and deserialize values. 

I think we don't need to rely on postcard, and we can stick with actual types.

As an example:

We already use `tables!` macro rule to define the `PerRegistryKeyspace` type with a separate field for each table. Instead of using `HashMap<RegistryKey, PostcardSerialized>` for each table, we could use `HashMap<RegistryKey, $type>` inside of the macro.

```rust
tables!(
    address: Address,
    asset_id: AssetId,
    contract_id: ContractId,
    script_code: Vec<u8>,
    predicate_code: Vec<u8>,
);
```
In this case, we move the decision about what serialization type to use to later stage.

Reply:
Got rid of the `PostcardSerialized` type in e0e505de032427bb941a48a9526558f5ba6f97e2